blocks_reexecutor/blocks_reexecutor.go

```diff
@@ -42,10 +42,9 @@ func (c *Config) Validate() error {
 }
 
 var DefaultConfig = Config{
-	Enable:          false,
-	Mode:            "random",
-	Room:            runtime.NumCPU(),
-	BlocksPerThread: 10000,
```

Comment:
we probably should keep the default for `BlocksPerThread: 10000` and in `Config.Validate` we should check that it is greater then zero - with that we should be able to skip the extra checks `c.BLocksPerThread` and hardcoding `10000` in more then one place

Reply:
hm... is there a reason not to have the default set by koanf?
If user does not set `blocks-per-thread` flag, DefaultConfig.BlocksPerThread will be used. If user set's it to 0, then we could return error from `Validate`. 
```
f.Uint64(prefix+".blocks-per-thread", DefaultConfig.BlocksPerThread, "minimum number of blocks to execute per thread. When mode is random this acts as the size of random block range sample")
```

Reply:
ah, sorry, I missed that `blocks-per-thread` has additional meaning if set to 0. 

Still, if I am not missing anything, we can use koanf to set the default if the flag is not used + add in the description something like `( 0 means equal split between reexecution threads)`. 

And I am still not a fan of modifying config structure here. I'd go with adding a blocksPerThread field to `BlocksReExecutor` and in `BlocksReExecutor.New` have something like:
```
blocksPerThread := config.BlocksPerThread
if config.BlocksPerThread == 0 {
    // split blocks equally to between threads
    blocksPerThread = (end - start) / uint64(c.Room)
    if blocksPerThread <= 0 {
        // only (end-start) threads will be started
        blocksPerThread = 1
    } 
}

return BlocksReExecutor {
     ...
     blocksPerThread: blocksPerThread,
     ...
}
```

Reply:
another argument for moving `blocksPerThread` field to `BlocksReExecutor` is that we already have analogous fields e.g. `BlocksReExector.startBlock` is set based on `Config.StartBlock` value